mechanics/physicell/include/physicell/agent_data.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+
+#include <common/base_agent_data.h>
+#include <common/types.h>
+
+namespace physicore::mechanics::physicell {
+
+template <template <typename...> typename ContainerType = std::vector>
+struct mechanical_agent_data
+{
+public:
+	physicore::base_agent_data_generic_storage<ContainerType>& base_data;
+
+	index_t agents_count = 0;
+	index_t dims = 3;
+	index_t agent_types_count = 0;
+	index_t substrates_count = 0;
+
+	// Geometry / kinematics
+	ContainerType<real_t> velocities;		   // current velocity per axis
+	ContainerType<real_t> previous_velocities; // previous velocity per axis
```

Comment:
these two are in plural but the rest is singular. lets rename to velocity and previous_velocity